operator/src/main/java/org/bf2/operator/managers/CapacityManager.java

```diff
@@ -0,0 +1,330 @@
+            }
+            return client.configMaps().withName(FLEETSHARD_RESOURCES).require();
+        }
+    }
+
+    ConfigMap checkForOrphans(ConfigMap configMap) {
+        boolean update = false;
+        Map<String, String> newData = new LinkedHashMap<>(configMap.getData());
+        for (Map.Entry<String, String> entry : configMap.getData().entrySet()) {
+            if (!entry.getKey().startsWith(MANAGED_KAFKA_PREFIX)) {
+                continue;
+            }
+            if (client.resources(ManagedKafka.class)
+                    .withLabel(ManagedKafka.ID, entry.getKey().substring(MANAGED_KAFKA_PREFIX.length()))
+                    .list()
+                    .getItems()
+                    .isEmpty()) {
+                update = true;
+                releaseResources(newData, entry.getKey());
+            }
+        }
+        if (update) {
+            configMap = new ConfigMapBuilder(configMap).withData(newData).build();
+            client.configMaps()
+                    .withName(FLEETSHARD_RESOURCES)
+                    .lockResourceVersion(configMap.getMetadata().getResourceVersion())
+                    .replace(configMap);
+        }
+        return configMap;
+    }
+
+    static Resources createResources(ManagedKafka mk, String currentProfile) {
+        Resources resources = new Resources();
+        resources.profile = currentProfile;
+        resources.units = Integer.parseInt(
+                OperandUtils.getOrDefault(mk.getMetadata().getLabels(), ManagedKafka.PROFILE_QUOTA_CONSUMED, "1"));
+        return resources;
+    }
+
+    public synchronized void releaseResources(ManagedKafka managedKafka) {
+        // use the latest
+        ConfigMap resourceConfigMap = client.configMaps().withName(FLEETSHARD_RESOURCES).get();
+        if (resourceConfigMap != null) {
+            String entryKey = getManagedKafkaKey(managedKafka);
+            if (releaseResources(resourceConfigMap.getData(), entryKey)) {
+                client.configMaps()
+                        .withName(FLEETSHARD_RESOURCES)
+                        .lockResourceVersion(resourceConfigMap.getMetadata().getResourceVersion())
+                        .replace(resourceConfigMap);
+            }
+        }
+    }
+
+    static String getManagedKafkaKey(ManagedKafka managedKafka) {
+        return "mk-" + managedKafka.getId();
+    }
+
+    private boolean releaseResources(Map<String, String> resourceData, String entryKey) {
+        String previous = resourceData.remove(entryKey);
+        if (previous != null) {
+            Resources resources = Serialization.unmarshal(previous, Resources.class);
+            resourceData.computeIfPresent(resources.profile,
+                    (k, v) -> String.valueOf(Integer.parseInt(v) - resources.units));
+            return true;
+        }
+        return false;
+    }
+
+    public synchronized Optional<OperandReadiness> claimResources(ManagedKafka managedKafka, String profile,
```

Comment:
Can the visibility of this method be reduced? Just to make it clear it's only used locally.

Reply:
Sure, no problem.